lib/spack/spack/build_systems/cargo.py

```diff
@@ -0,0 +1,247 @@
+        use Rust's LLVM version in place of the clang version."""
+
+        return RustQuery(self.rustc).target_cpu(self.spec)
+
+    @property
+    def manifest_path(self):
+        return join_path(self.stage.source_path, self.cargo_manifest)
+
+    @property
+    def metadata(self):
+        """Returns the metadata description of the crate.
+
+        Used to determine the installable targets in the crate"""
+
+        feature_args = self._feature_args()
+
+        return json.loads(inspect.getmodule(self).cargo(
+            'metadata', '--manifest-path', self.manifest_path,
+            '--format-version', '1',
+            *feature_args,
+            output=str))
+
+    @property
+    def target_path(self):
+        return join_path(
+            self.stage.source_path, 'target',
+            target_triple_for_spec(self.spec),
+            self.spec.variants['build_type'].value)
+
+    def do_stage(self, mirror_only=False):
+        """Stages and sets the config for chosen build settings"""
+        super(CargoPackage, self).do_stage(mirror_only)
+
+        rustflags = []
+
+        target = target_triple_for_spec(self.spec)
+
+        target_cpu = self.target_cpu
+        if target_cpu:
+            rustflags.extend(["-C", "target-cpu={0}".format(target_cpu)])
+
+        # Put config in spack-src so that it does not overwrite the config
+        # from vendoring dependencies
+        mkdirp(join_path(self.stage.source_path, '.cargo'))
+        config_path = join_path(self.stage.source_path, '.cargo/config')
+
+        build_type = self.spec.variants['build_type'].value
+        lto = self.spec.variants['lto'].value
+
+        profile = 'dev' if build_type == 'debug' else build_type
+        lto = 'off' if lto == 'none' else lto
+
+        with open(config_path, 'w') as f:
+            f.write("""
+[build]
+target = "{target}"
+rustflags = "{flags}"
+
+[profile.{profile}]
+rpath = true
+lto = "{lto}"
+""".format(
+                target=target,
+                flags=" ".join(rustflags),
+                profile=profile,
+                lto=lto
+            ))
+
+    def workspace_targets(self):
+        metadata = self.metadata
+        for p in metadata["packages"]:
+            for w in metadata["workspace_members"]:
+                if p["id"] == w:
+                    for t in p["targets"]:
+                        yield t
+
+    def cargo_features(self):
+        """Features to build cargo project with. Overridable by package."""
+        return None
+
+    def _feature_args(self):
+        """Returns the args associated with cargo_features"""
+        features = self.cargo_features()
+        if features is None:
+            return []
+
+        args = ["--no-default-features"]
+
+        for feature in features:
+            args += ["--features", feature]
+
+        return args
+
+    def build(self, spec, prefix):
+        """cargo build"""
+        jobs = spack.config.get('config:build_jobs') if self.parallel else 1
```

Comment:
Use `make_jobs` here, you'll probably need `inspect` to access it. See other build systems for how to do this.